gdal/port/userfaultfd.h

```diff
@@ -0,0 +1,42 @@
+/******************************************************************************
+ *
+ * Name:     userfaultfd.h
+ * Project:  CPL - Common Portability Library
+ * Purpose:  Use userfaultfd and VSIL to service page faults
+ * Author:   James McClain, <james.mcclain@gmail.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2018, James McClain <james.mcclain@gmail.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#ifndef __USERFAULTFD__
+#define __USERFAULTFD__
+
+#include <fcntl.h>
```

Comment:
to conditionlize with the minimum kernel version needed and for other OS

Reply:
I am doing it at runtime with `uname(2)`.  I will make sure that this code is never compiled on systems that do not have the `linux/userfaultfd.h` header.